### DISCUSSION TITLE
No literal value renderer is available for literal value "'english'" with datatype REGCONFIG

### DISCUSSION THREAD

**Discussion by auxsvr** [sqlalchemy/sqlalchemy#9910] [Usage Questions] [answered]
With a model used with SA 1.4 defined as:
```python
class M(Base):
    description = Column(String())
    @declared_attr
    def __table_args__(cls):
        return (
            Index("ix_description", func.to_tsvector("english", cls.description), postgresql_using="gin", postgresql_ops={"description": "gin_trgm_ops"}),
            super(cls, cls).__table_args__,
        )
```
alembic with SA 2.0.15 throws:

> sqlalchemy.exc.CompileError: No literal value renderer is available for literal value "'english'" with datatype REGCONFIG

The solution in https://stackoverflow.com/q/75292767/1800194 works, but in some cases no cast is required and in others the special cast in the answer must be done, which looks inconsistent, perhaps indicative of a bug.

**Comment by zzzeek** [comment, marked as answer]
render the to_tsvector like this:

    func.to_tsvector(literal_column("'english'"), cls.description)

the datatype here does not have a string render for literal_binds

edited: quotes

**Reply by LLYX** [reply]
When I do this, the compile error disappears but now I get:
`sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedColumn) column "english" does not exist`

Could I get some advice on why this is happening?

**Reply by LLYX** [reply]
Found a solution, instead of using literal_column, I used:
`func.to_tsvector(bindparam("ts_config", ts_config, type_=Text), field)`

**Reply by zzzeek** [reply]
this would indicate you missed the extra set of quotes, note the double quotes: `literal_column('"english"')`

**Reply by LLYX** [reply]
Hmm, I had a variable with the value "english", and I used it using an f-string that had the extra set of quotes.

`ts_config = "english"`
`func.to_tsvector(literal_column(f'"{ts_config}"', field)`

Not sure if this is a known limitation of f-strings but I thought that should have worked, yet I still got that ProgrammingError. Does my current solution make sense or would it lead to other unintended consequences?

**Reply by zzzeek** [reply]
Postgresql considers double quotes to be identifier quoting:

```
# no such column "english"
SELECT "english"
```

single quotes are literals:

```
# returns 'english'
SELECT 'english'
```

therefore pass as single quotes:

```py
from sqlalchemy import *

t = Table(
    "t",
    MetaData(),
    Column("q", Integer, primary_key=True),
    Column("data", String),
)
engine = create_engine("postgresql://scott:tiger@localhost/test", echo=True)
with engine.connect() as conn:
    t.create(conn)

    ts_config = "english"
    conn.execute(
        select(func.to_tsvector(literal_column(f"'{ts_config}'"), t.c.data))
    )

```

updated my answer above
